watchtower/__init__.py

```diff
@@ -135,11 +135,9 @@ def __init__(self, log_group=__name__, stream_name=None, use_queues=True, send_i
         self.max_batch_size = max_batch_size
         self.max_batch_count = max_batch_count
         self.max_message_size = max_message_size
-        self.queues, self.sequence_tokens = {}, {}
-        self.threads = []
-        self.creating_log_stream, self.shutting_down = False, False
         self.create_log_stream = create_log_stream
         self.log_group_retention_days = log_group_retention_days
+        self._init_state()
```

Comment:
This will be called with the parent's `__init__` which creates its lock. Since it's ok to init the state twice I made it more obvious it's being called by calling it directly. I can add a comment on `super().__init__` if that seems better.